src/app/home/home.page.ts

```diff
@@ -74,102 +76,101 @@ export class HomePage implements OnInit {
       }
     });
   }
-  createLetters() {
+
+  private createLetters() {
     this.letters = Object.keys(this.groupedCountries).sort();
   }
 
- 
+  private updatedLoadingState(item: Country): void {
+    this.isLoaded = true;
+    this.isLoading = false;
+    this.loadedItem = item;
+    this.loadingItem = null;
+  }
+
+  private addToFavorites(item: Country): void {
+    if (!this.favourites.includes(item))
+      this.favourites.push(item);
+  }
+
+  private removeFromFavorites(item: Country): void {
+    this.favourites = this.favourites.filter(c => c !== item);
+  }
+
+  private updateView(): void {
+    if (!this.isHomePage) {
+      this.groupedCountries = {};
+      this.countries = this.favourites;
+      this.groupCountries();
+      this.createLetters();
+    } else {
+      this.groupedCountries = this.baseBackupGroupedCountries;
+    }
+  }
+
+  filterCountries(searchBarValue: string | undefined | null): void {
```

Comment:
1-  utilize logical operators and conditional expressions effectively.
2- Grouped Countries and Letters